core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -556,6 +556,7 @@ union SharedAllocationTracker {
 
   KOKKOS_FORCEINLINE_FUNCTION
   SharedAllocationTracker& operator=(SharedAllocationTracker&& rhs) {
+    if (&rhs == this) return *this;
```

Comment:
Is there any reason we aren't just calling `kokkos::swap(m_record_bits, rhs.m_record_bits);`?

Reply:
not sure but if we want to change move behavior of SharedAllocationTracker we should probably do it separately. This just fixes the problem that self assignment/move is broken and would result in screwed up reference count. We didn't notice before because View was using the tracker only through a wrapping ViewTracker object, while now we use it more directly.